lib/wibox/widget/slider.lua

```diff
@@ -295,8 +302,19 @@ function slider:draw(_, cr, width, height)
         or beautiful.slider_bar_border_width
         or properties.bar_border_width
 
-    bar_shape(cr, width - x_offset - right_margin, bar_height or height)
 
+    if bar_bg then cr:set_source(color(bar_bg)) end
+    bar_shape(cr, width - x_offset - right_margin, bar_height or height)
+    if bar_bg then
+        if bar_border_width == 0 then
+            cr:fill()
+        else
+            cr:fill_preserve()
```

Comment:
If there is a border, `cr:fill_preserve()` is called. Due to that the area will be repainted with `bar_color` (if `bar_color` is set).